ethereumj-core/src/main/java/org/ethereum/listener/CompositeEthereumListener.java

```diff
@@ -50,10 +51,14 @@ public String toString() {
         }
     }
 
-    @Autowired
-    EventDispatchThread eventDispatchThread = EventDispatchThread.getDefault();
-    
-    protected List<EthereumListener> listeners = new CopyOnWriteArrayList<>();
+    private final Executor executor;
+    private final List<EthereumListener> listeners;
+
+
+    public CompositeEthereumListener(Executor executor) {
```

Comment:
Why do we have anything to update in deprecated class?

Reply:
Firstly field injection is bad practice cause it hides component dependencies. But in this case it's not critical because class marked as deprecated.
I moved bean definition to java config to highlight that old and new components use the same `Executor`. And then both `Publisher` and deprecated `CompositeEthereumListener` injects into `BackwardCompatibilityEthereumListenerProxy`. IMHO it's more obvious and helps avoid dependency mess. 

This also applies to the previous comment.